src/core/lib/transport/metadata.h

```diff
@@ -132,17 +134,114 @@ void* grpc_mdelem_get_user_data(grpc_mdelem md, void (*if_destroy_func)(void*));
 void* grpc_mdelem_set_user_data(grpc_mdelem md, void (*destroy_func)(void*),
                                 void* user_data);
 
+void grpc_mdelem_trace_ref(void* md, const grpc_slice& key,
```

Comment:
are these two functions used on non-debug builds? if not I'd remove them?

Reply:
Moved the definition so it's guarded by #ifndef NDEBUG.